From a18e130d6caff08ebc57a665e7ff8e4aa3c7cb3e Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Mon, 12 Oct 2026 17:10:27 +0100
Subject: [PATCH] dt-bindings: media: rockchip: Add Rockchip RKVENC v1 encoder
 binding

Add the device tree binding for the Rockchip RKVENC v1 video encoder
found in the RV1126/RV1109 ("vepu541") and RK3566/RK3568 ("vepu540")
SoCs. The block encodes H.264 (AVC) and H.265 (HEVC) from raw
YUV/RGB frames and reaches memory through an IOMMU.

Document the two compatible strings, the register space, the single
interrupt, the axi/ahb/core clock and reset triplets, the power
domain and the IOMMU phandle.

Assisted-by: LLM
Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 .../media/rockchip,rv1126-rkvenc.yaml         | 102 ++++++++++++++++++
 1 file changed, 102 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/rockchip,rv1126-rkvenc.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/media/rockchip,rv1126-rkvenc.yaml
@@ -0,0 +1,102 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/rockchip,rv1126-rkvenc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Rockchip RKVENC v1 H.264/HEVC video encoder
+
+maintainers:
+  - Daniel Golle <daniel@makrotopia.org>
+
+description:
+  The RKVENC v1 encoder found in the RV1109, RV1126, RK3566 and RK3568
+  encodes H.264 and HEVC from raw YUV or RGB frames. It reads and writes
+  memory through an IOMMU with two MMU instances.
+
+properties:
+  compatible:
+    enum:
+      - rockchip,rk3568-rkvenc
+      - rockchip,rv1126-rkvenc
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+  clocks:
+    items:
+      - description: AXI master clock
+      - description: AHB register interface clock
+      - description: Encoder core clock
+
+  clock-names:
+    items:
+      - const: axi
+      - const: ahb
+      - const: core
+
+  assigned-clocks: true
+
+  assigned-clock-rates: true
+
+  resets:
+    items:
+      - description: AXI reset
+      - description: AHB reset
+      - description: Encoder core reset
+
+  reset-names:
+    items:
+      - const: axi
+      - const: ahb
+      - const: core
+
+  power-domains:
+    maxItems: 1
+
+  iommus:
+    maxItems: 1
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - clocks
+  - clock-names
+  - resets
+  - reset-names
+  - power-domains
+  - iommus
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/rockchip,rv1126-cru.h>
+    #include <dt-bindings/interrupt-controller/arm-gic.h>
+    #include <dt-bindings/power/rockchip,rv1126-power.h>
+
+    venc_mmu: iommu@ffbb0800 {
+        compatible = "rockchip,rv1126-iommu";
+        reg = <0xffbb0800 0x100>;
+        #iommu-cells = <0>;
+    };
+
+    video-codec@ffbb0000 {
+        compatible = "rockchip,rv1126-rkvenc";
+        reg = <0xffbb0000 0xf00>;
+        interrupts = <GIC_SPI 68 IRQ_TYPE_LEVEL_HIGH>;
+        clocks = <&cru ACLK_VENC>, <&cru HCLK_VENC>, <&cru CLK_VENC_CORE>;
+        clock-names = "axi", "ahb", "core";
+        assigned-clocks = <&cru ACLK_VENC>, <&cru CLK_VENC_CORE>;
+        assigned-clock-rates = <297000000>, <396000000>;
+        resets = <&cru SRST_VENC_A>, <&cru SRST_VENC_H>,
+                 <&cru SRST_VENC_CORE>;
+        reset-names = "axi", "ahb", "core";
+        iommus = <&venc_mmu>;
+        power-domains = <&power RV1126_PD_VEPU>;
+    };
+...
